From 70ac1232e34717aa0dbea51251e0530a6456358d Mon Sep 17 00:00:00 2001
From: Christophe Kerello <christophe.kerello@foss.st.com>
Date: Wed, 20 Apr 2022 11:34:34 +0200
Subject: [PATCH 163/841] feat(stm32mp2): use 4-bytes fast read opcode

On MP25, all SPI NOR memories used supports 4-bytes fast read opcode.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 2e2154a55b219288aa024c3b4668b873d81639f5 feat(stm32mp2): use 4-bytes fast read opcode]
Change-Id: I5b2ca278658b35fdd3af60ea71e7a2d3d6bd3776
Signed-off-by: Christophe Kerello <christophe.kerello@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/248753
Reviewed-by: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Christophe KERELLO <christophe.kerello@st.com>
Reviewed-by: Yann GAUTIER <yann.gautier@foss.st.com>
Tested-by: Christophe KERELLO <christophe.kerello@st.com>
---
 plat/st/stm32mp2/stm32mp2_boot_device.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/plat/st/stm32mp2/stm32mp2_boot_device.c b/plat/st/stm32mp2/stm32mp2_boot_device.c
index 67553b0f2..55a985c3f 100644
--- a/plat/st/stm32mp2/stm32mp2_boot_device.c
+++ b/plat/st/stm32mp2/stm32mp2_boot_device.c
@@ -156,6 +156,16 @@ int plat_get_nor_data(struct nor_device *device)
 {
 	device->size = SZ_64M;
 
+	zeromem(&device->read_op, sizeof(struct spi_mem_op));
+	device->read_op.cmd.opcode = SPI_NOR_OP_READ_FAST_4B;
+	device->read_op.cmd.buswidth = SPI_MEM_BUSWIDTH_1_LINE;
+	device->read_op.addr.nbytes = 4U;
+	device->read_op.addr.buswidth = SPI_MEM_BUSWIDTH_1_LINE;
+	device->read_op.dummy.nbytes = 1U;
+	device->read_op.dummy.buswidth = SPI_MEM_BUSWIDTH_1_LINE;
+	device->read_op.data.buswidth = SPI_MEM_BUSWIDTH_1_LINE;
+	device->read_op.data.dir = SPI_MEM_DATA_IN;
+
 	return 0;
 }
 #endif
-- 
2.47.2

